pkg/pillar/devicenetwork/dns.go

```diff
@@ -131,6 +131,10 @@ func ResolveWithPortsLambda(domain string,
 				copy(srcIPCopy, srcIP)
 				countDNSRequests++
 				go func(dnsIP, srcIP net.IP) {
+					defer func() {
+						wg.Done()
+						<-work
+					}()
```

Comment:
I think that the following lines (
```
select {
    case work <- struct{}{}:
	// if writable, means less than dnsMaxParallelRequests goroutines are currently running
}
```
)
could be simplified to
```
work <- struct{}{}
```

Reply:
done